File: chap14/fig14-5-right.R
library(dplyr)
library(tidyr)
library(ggplot2)
library(latex2exp)

N <- 50
K <- 6
I <- 120
fit_MAP <- readRDS('output/result-model14-3.RDS')
theta_est <- fit_MAP$draws(format='df') %>% 
  select(starts_with('theta[')) %>% 
  pivot_longer(everything(), names_to='Parameter') %>% 
  extract(col='Parameter', into=c('PersonID', 'Pattern'), regex='theta\\[(\\d+),(\\d+)\\]', convert=TRUE) %>% 
  mutate(PersonID_label=paste0('n = ', PersonID))

p <- ggplot(data=subset(theta_est, PersonID %in% c(1,50)), aes(x=Pattern, y=value)) +
  theme_bw(base_size=18) +
  facet_wrap(~PersonID_label, nrow=2) +
  coord_flip() +
  scale_x_reverse(breaks=1:6) +
  geom_bar(stat='identity') +
  labs(y=TeX('$\\theta\\[n,k\\]$'))
ggsave(p, file='output/fig14-5-right.png', dpi=300, w=5, h=5)

File: chap14/fig14-5-right.py
import pandas
import cmdstanpy
from plotnine import ggplot, aes, theme_bw, facet_wrap, coord_flip, scale_x_reverse, geom_bar, labs

d = pandas.read_csv('input/data-matrix-decomp.csv')
N = 50
K = 6
I = 120

fit_MAP = cmdstanpy.from_csv('output/result-model14-3')
theta_est = pandas.DataFrame(fit_MAP.stan_variable('theta'), columns=range(1, K+1))
theta_est = pandas.concat([pandas.DataFrame({'PersonID':range(1, N+1)}), theta_est], axis=1)
d_plot = pandas.melt(theta_est, id_vars='PersonID', var_name='Pattern')
d_plot['Pattern'] = d_plot.Pattern.astype('int32')

p = (ggplot(d_plot.query('PersonID in [1,50]').reset_index(), aes('Pattern', 'value'))
    + theme_bw(base_size=18)
    + facet_wrap('~ PersonID', nrow=2)
    + coord_flip()
    + scale_x_reverse(breaks=range(1,K+1))
    + geom_bar(stat='identity')
    + labs(y='$\theta[n,k]$')
)
p.save(filename='output/fig14-5-right.py.png', dpi=300, width=5, height=5)
